webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/MockServices.scala

```diff
@@ -2837,7 +2801,8 @@ class MockNodeGroups(mockNodes: MockNodes, mockGlobalParam: MockGlobalParam) {
         None,
         None
       )
-      .getOrElse(throw new RuntimeException(s"error group prop: jsonParam")) // for test
+      .getOrElse(throw new RuntimeException(s"error group prop: jsonParam")), // for test
+    GroupProperty("hiddenProp", GitVersion.DEFAULT_REV, "hidden prop value".toConfigValue, None, None).withVisibility(Hidden)
```

Comment:
I do not really understand why there is no `visibility: Visibility` in the constructor of `NodeProperty`, `GroupProperty`, even if there already is a `.withVisibility` method. Is it to avoid propagating the changes in plugins and APIs everywhere ?

Reply:
Yes, and for now, it looks like an exceptionnal attribute. 
If it becomes common in the future (possible, since properties are so powerful, we are likely to want to use them more and more, for thing that will be intenal to rudder behavior), we will change that.